src/spock_recovery.c

```diff
@@ -947,3 +1044,164 @@ spock_clone_recovery_slot_sql(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(HeapTupleGetDatum(tuple));
 }
+
+/*
+ * spock_create_rescue_subscription_sql
+ *
+ * Create a temporary rescue subscription from the current (lagging) node to a
+ * cloned recovery slot on a more advanced peer. The subscription uses the
+ * cloned slot, optionally skips already processed transactions, and stops at a
+ * known-safe LSN/timestamp to allow automatic cleanup.
+ */
+Datum
+spock_create_rescue_subscription_sql(PG_FUNCTION_ARGS)
+{
+	Name		target_name = PG_GETARG_NAME(0);
```

Comment:
Can we use `pg_copy_logical_replication_slot` and `pg_replication_slot_advance` functions perhaps?